internal/credential/vault/testing.go

```diff
@@ -0,0 +1,853 @@
+// The orphan option is true by default.
+func TestOrphanToken(b bool) TestOption {
+	return func(t *testing.T, o *testOptions) {
+		t.Helper()
+		o.orphan = b
+	}
+}
+
+// TestPeriodicToken sets the token periodic option to b.
+// The periodic option is true by default.
+func TestPeriodicToken(b bool) TestOption {
+	return func(t *testing.T, o *testOptions) {
+		t.Helper()
+		o.periodic = b
+	}
+}
+
+// TestRenewableToken sets the token renewable option to b.
+// The renewable option is true by default.
+func TestRenewableToken(b bool) TestOption {
+	return func(t *testing.T, o *testOptions) {
+		t.Helper()
+		o.renewable = b
+	}
+}
+
+func (v *TestVaultServer) client(t *testing.T) *client {
+	t.Helper()
+	return v.clientUsingToken(t, v.RootToken)
+}
+
+func (v *TestVaultServer) clientUsingToken(t *testing.T, token string) *client {
+	t.Helper()
+	require := require.New(t)
+	conf := &clientConfig{
+		Addr:       v.Addr,
+		Token:      TokenSecret(token),
+		CaCert:     v.CaCert,
+		ClientCert: v.ClientCert,
+		ClientKey:  v.ClientKey,
+	}
+
+	client, err := newClient(conf)
+	require.NoError(err)
+	require.NotNil(client)
+	require.NoError(client.ping())
+	return client
+}
+
+// CreateToken creates a new Vault token by calling /auth/token/create on v
+// using v.RootToken. It returns the vault secret containing the token and
+// the token itself. See
+// https://www.vaultproject.io/api-docs/auth/token#create-token.
+func (v *TestVaultServer) CreateToken(t *testing.T, opt ...TestOption) (*vault.Secret, string) {
+	t.Helper()
+	require := require.New(t)
+	opts := getTestOpts(t, opt...)
+
+	var period string
+	if opts.periodic {
+		period = "24h"
+		if deadline, ok := t.Deadline(); ok {
+			period = time.Until(deadline).String()
+		}
+	}
+	req := &vault.TokenCreateRequest{
+		DisplayName: t.Name(),
+		NoParent:    opts.orphan,
+		Renewable:   &opts.renewable,
+		Period:      period,
+		Policies:    opts.policies,
+	}
+	vc := v.client(t).cl
+	secret, err := vc.Auth().Token().Create(req)
+	require.NoError(err)
+	require.NotNil(secret)
+	token, err := secret.TokenID()
+	require.NoError(err)
+	require.NotNil(token)
+
+	return secret, token
+}
+
+// LookupToken calls /auth/token/lookup on v for the token. See
+// https://www.vaultproject.io/api-docs/auth/token#lookup-a-token.
+func (v *TestVaultServer) LookupToken(t *testing.T, token string) *vault.Secret {
+	t.Helper()
+	require := require.New(t)
+	vc := v.client(t).cl
+	secret, err := vc.Auth().Token().Lookup(token)
+	require.NoError(err)
+	require.NotNil(secret)
+	return secret
+}
+
+// VerifyTokenInvalid calls /auth/token/lookup on v for the token. It expects the lookup to
+// fail with a StatusForbidden.  See
+// https://www.vaultproject.io/api-docs/auth/token#lookup-a-token.
+func (v *TestVaultServer) VerifyTokenInvalid(t *testing.T, token string) {
+	t.Helper()
+	require, assert := require.New(t), assert.New(t)
+	vc := v.client(t).cl
+	secret, err := vc.Auth().Token().Lookup(token)
```

Comment:
Same comment.

Reply:
This call is also part of the test harness and it is currently only used in tests that revoke tokens and need to verify the token has been revoked.